collagraph/collagraph.py

```diff
@@ -551,17 +529,42 @@ def is_equivalent_event_handler(val, other, key):
             for event_type, val in events_to_add.items():
                 self.renderer.add_event_listener(dom, event_type, val)
 
-        # Climb up the tree to find first component to mark as updated
-        if events_to_remove or events_to_add or attrs_to_remove or attrs_to_update:
-            parent = fiber.parent
-            component = None
-            while parent:
-                component = parent.component
-                if component:
-                    # Mark the fiber as updated
-                    parent.updated = True
-                    break
-                parent = parent.parent
+            # Climb up the tree to find first component to mark as updated
+            if events_to_remove or events_to_add or attrs_to_remove or attrs_to_update:
+                parent = fiber.parent
+                component = None
+                while parent:
+                    component = parent.component
+                    if component:
+                        # Mark the fiber as updated
+                        parent.updated = True
+                        break
+                    parent = parent.parent
+
+        if fiber.component:
+            for event_type, val in events_to_remove.items():
+                fiber.component.remove_event_handler(event_type, val)
+
+            for event_type, val in events_to_add.items():
+                fiber.component.add_event_handler(event_type, val)
+
+
+def is_event(key):
+    return key.startswith("on_")
+
+
+def key_to_event(key):
+    # Events start with `on_`
+    return key[3:].lower()
```

Comment:
or `@` right?

Reply:
`@` is currently only valid in templates, where the `@` is replaced by `_on` during 'compilation'. We could think about also supporting `@` in other places though. What do you think?